hermes/probe/delete/delete_file_test.go

```diff
@@ -0,0 +1,178 @@
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/mock"
+	"google.golang.org/api/iterator"
+
+	metricpb "github.com/google/cloudprober/metrics/proto"
+	monitorpb "github.com/googleinterns/step224-2020/config/proto"
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	journalpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+// genTestConfig generates a test HermesProbeDef proto config.
+// Arguments:
+//	- name: pass the name for this probe instance.
+// Returns:
+//	- hermesExtension: returns the HermesProbeDef extension.
+func genTestConfig(name string) *monitorpb.HermesProbeDef {
+	hermesExtension := &monitorpb.HermesProbeDef{
+		ProbeName: proto.String(name),
+		Targets: []*monitorpb.Target{
+			genTargetPb(),
+		},
+		TargetSystem: monitorpb.HermesProbeDef_GCS.Enum(),
+		IntervalSec:  proto.Int32(3600),
+		TimeoutSec:   proto.Int32(60),
+		ProbeLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1,0.2,0.4,0.6,0.8,1.6,3.2,6.4,12.8,1000",
+			},
+		},
+		ApiCallLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.000000002,0.000000004,0.000000008,0.000000016,0.00000032,0.000000064,0.000000128,100",
+			},
+		},
+	}
+	return hermesExtension
+}
+
+// getTargetPb generates a target proto config.
+// Returns:
+//	- target: target proto config
+func genTargetPb() *monitorpb.Target {
+	return &monitorpb.Target{
+		Name:                   "hermes",
+		TargetSystem:           monitorpb.Target_GOOGLE_CLOUD_STORAGE,
+		TotalSpaceAllocatedMib: int64(100),
+		BucketName:             bucketName,
+	}
+}
+
+const (
+	firstID    = int32(1)
+	lastID     = int32(50)
+	contents   = "abc123"
+	hash       = "6367c48dd193d56ea7b0baad25b19455e529f5ee"
+	bucketName = "test_bucket_5"
+)
+
+// genTestTarget generates an initialised test Target struct.
+func genTestTarget(cfg *monitorpb.HermesProbeDef, t *testing.T) *probe.Target {
+	filenames := make(map[int32]string)
+
+	for i := firstID; i <= lastID; i++ {
+		var id string
+		if i <= 9 {
+			id = fmt.Sprintf("0%d", i)
+		} else {
+			id = fmt.Sprintf("%d", i)
+		}
+		filenames[i] = fmt.Sprintf("Hermes_%s_%s", id, hash)
+	}
+
+	metrics, err := m.NewMetrics(cfg, genTargetPb())
+	if err != nil {
+		t.Fatalf("could not initialise metrics using config and target provided, err: %v", err)
+	}
+
+	return &probe.Target{
+		Target: cfg.GetTargets()[0],
+		Journal: &journalpb.StateJournal{
+			Intent:    &journalpb.Intent{},
+			Filenames: filenames,
+		},
+		LatencyMetrics: metrics,
+	}
+}
+
+// setupMockSystem sets up the mock storage system through the fake client.
+func setupMockSystem(ctx context.Context, t *testing.T) stiface.Client {
```

Comment:
s/Mock/Fake/, here and elsewhere.

Reply:
Thank you, I've changed this as requested.